src/render/painter.js

```diff
@@ -746,6 +748,21 @@ class Painter {
         ext.beginQueryEXT(ext.TIME_ELAPSED_EXT, layerTimer.query);
     }
 
+    gpuTimingGlobeStart() {
+        if (this.options.gpuTimingGlobe) {
+            const ext = this.context.extTimerQuery;
+            if (!this.globeGpuTimeQuery)
+                this.globeGpuTimeQuery = ext.createQueryEXT();
+            ext.beginQueryEXT(ext.TIME_ELAPSED_EXT, this.globeGpuTimeQuery);
+        }
+    }
+
+    gpuTimingGlobeEnd() {
+        if (!this.options.gpuTimingGlobe) return;
+        const ext = this.context.extTimerQuery;
+        ext.endQueryEXT(ext.TIME_ELAPSED_EXT);
+    }
+
```

Comment:
We already have code in place for measuring frame times. Do we need a separate implementation for this one?

Reply:
More generally, maybe it should be more tied to drawing any deferred raster (e.g. including terrain too)? [here](https://github.com/mapbox/mapbox-gl-js/pull/11604/files#diff-0b3f9fb815ac75ccdb17a0b25c8dfbe915949a8dc84ea322aa209f5c07880ba0R265) we inject timing queries when drawing globe, but we could move that one level up and just have it over `drawTerrainRaster`. 

The query/function name would be associated to `gpu-timing-deferred-raster`.

Reply:
More generally, maybe it should be more tied to drawing any deferred raster (e.g. including terrain too)? [here](https://github.com/mapbox/mapbox-gl-js/pull/11604/files#diff-0b3f9fb815ac75ccdb17a0b25c8dfbe915949a8dc84ea322aa209f5c07880ba0R265) we inject timing queries when drawing globe, but we could move that one level up and just have it over `drawTerrainRaster`. 

The query/function name would be associated to `gpu-timing-deferred-raster`.

Reply:
I was actually considering this but took this approach in the end. I think we could have it more generic which could be used to also measure terrain related timings.